### PR TITLE
restructing of the structs, to close #150

### LINKED ISSUE #150: Optimize struct ordering for gas savings
BLUF: Saving about 128 bytes with this change per Order. I also examined Peer and Indexer but did not find areas for optimization.

Solidity works in 32 bytes slots and so if we can fit fixed params within a single slot, re-arranged the params to do just that.

Current
```
  struct Signature {
    address signer;       // 20 byte
    uint8 v;              // 1 byte 
    bytes32 r;            // 32 bytes
    bytes32 s;            // 32 bytes
    bytes1 version;       // bytes 1
  }
  4 - 32 bytes

  struct Party {
    address wallet;       // 20 byte
    address token;        // 20 byte
    uint256 param;        // 32 bytes
    bytes4 kind;          // 4 bytes
  }

  4 - 32 bytes 

  struct Order {
    uint256 nonce;        // 32 bytes
    uint256 expiry;       // 32 bytes
    Party maker;          // 4 - 32 bytes
    Party taker;          // 4 - 32 bytes
    Party affiliate;      // 4 - 32 bytes
    Signature signature;  // 4 - 32 bytes
  }
  18 - 32 bytes
```
Proposed
```
  struct Signature {
    address signer;       //20 bytes
    bytes1 version;       //1 bytes
    uint8 v;              //1 byte
    bytes32 r;            //32 bytes
    bytes32 s;            //32 byetes
  }
  3 - 32 bytes

  struct Party {
    bytes4 kind;          // 4 bytes
    address wallet;       // 20 bytes
    address token;        // 20 bytes
    uint256 param;        // 32 bytes
    
  }
  3 - 32 bytes

  struct Order {
    uint256 nonce;        // 32 bytes
    uint256 expiry;       // 32 bytes
    Party maker;          // 3 - 32 bytes
    Party taker;          // 3 - 32 bytes
    Party affiliate;      // 3 - 32 bytes
    Signature signature;  // 3 - 32 bytes
  }
14 - 32 bytes
```

### PR DESCRIPTION
**Pull request by poiresel**
## Description

Quick description: 
Updated struct for Party and Signature.

- [X] Typo/small tweaks

## Changes

Two primary changes in Types.sol that resulted in Party struct and Signature struct to have the below ordering. Longer explanation found in #150. 

  ```
struct Party {
    bytes4 kind;          // Interface ID of the token
    address wallet;       // Wallet address of the party
    address token;        // Contract address of the token
    uint256 param;        // Value (ERC-20) or ID (ERC-721)
  }

  struct Signature {
    address signatory;    // Address of the wallet used to sign
    bytes1 version;       // EIP-191 signature version
    uint8 v;              // `v` value of an ECDSA signature
    bytes32 r;            // `r` value of an ECDSA signature
    bytes32 s;            // `s` value of an ECDSA signature
  }
```

## Tests

All tests pass

## Test Coverage

For Types.sol
```
--------------|----------|----------|----------|----------|----------------|
File          |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
--------------|----------|----------|----------|----------|----------------|
 contracts/   |      100 |      100 |      100 |      100 |                |
  Imports.sol |      100 |      100 |      100 |      100 |                |
  Types.sol   |      100 |      100 |      100 |      100 |                |
--------------|----------|----------|----------|----------|----------------|
All files     |      100 |      100 |      100 |      100 |                |
--------------|----------|----------|----------|----------|----------------|
```

Coverage has not changed

### CODE REVIEW

**Comment by ejwessel** [comment]
Do we have the comparison of gas savings from before and after?

File: packages/order-utils/test/signatures.js [hunk 1]
```diff
@@ -39,15 +39,19 @@ describe('Signatures', async () => {
       signerPrivKey,
       swapAddress
     )
+
+    console.log(signature.r.toString('hex'))
+    console.log(signature.s.toString('hex'))
```
Comment (dmosites) [inline packages/order-utils/test/signatures.js]: Please remove

File: packages/order-utils/test/signatures.js [hunk 2]
```diff
@@ -78,15 +82,18 @@ describe('Signatures', async () => {
...
```
Comment (dmosites) [inline packages/order-utils/test/signatures.js]: Please remove

File: packages/order-utils/test/signatures.js (hunk 1 above)
Comment (poiresel) [reply packages/order-utils/test/signatures.js]: done!

File: packages/order-utils/test/signatures.js (hunk 2 above)
Comment (poiresel) [reply packages/order-utils/test/signatures.js]: done!

**Pull request merged**